README.md

```diff
@@ -1,22 +1,336 @@
+right? When the web was first becoming more dynamic, and web
+services/APIs were in their nascent stages, XML was the lingua franca of
+inter-system communication. That's where **AJAX** came from:
+*Asynchronous JavaScript and XML*.
+
+So this is a `Post` instance serialized to XML. The XML describes the
+state of the object, and we can find where to pull out individual bits
+of data, or even reconstruct the object in memory on a new system. In
+this way, XML was a great tool.
+
+But XML had its problems as well. Just as browsers have a hard time
+agreeing on how to parse tag-based HTML, parsing XML is a chore. An XML
+parser has to essentially *read* through a document, finding the start
+and end tags of nodes and deciding what to do with the data. Searching
+XML documents with [XPATH](https://en.wikipedia.org/wiki/XPath) is
+cumbersome, and constructing an XML document can be a tedious exercise
+in string building and tag matching.
+
+In addition, XML has the potential to inflate the size of the result
+exponentially. Look at the `<authorId>` tag in the example above.
+There's 20 bytes of metadata per one byte of data. Even in a system with
+hundreds of authors, that's an inefficient way of representing things.
+As the Internet grew and the demands on API speed and efficiency,
+especially as SmartPhones came into play over cellular networks, the
+size of the data being passed from API to consumer became *important*.
+
+### JSON
+
+JSON stands for "JavaScript Object Notation", and was created as a way
+to use JavaScript to not only consume data from an API, but also
+serialize that data for consumption. Since so much data was being used
+in AJAX calls by the mid 2000s, it only made sense to start offering
+that data in the native language of the calling code.
+
+Beyond that, there's significant advantages to JSON. If we serialize
+that same `Post` to JSON, it looks like this:
+
+```javascript
+{
+  "id": 1,
+  "title": "This is a blog post",
+  "description": "Lorem ipsum sit amet dolor...",
+  "author": {
+    "id":2,
+    "name": "Stephen King",
+  }
+}
+```
+
+This is still very readable, and it's easy to see what represents what,
+but just in this one example we've shaved almost 50 bytes from the
+string. That might not seem like much, but it's an almost 25%
+improvement in size, which, when stacked up against your mobile data
+plan, means 25% more times you get to obsessively check your Facebook
+likes.
+
+![snap](http://giphy.com/gifs/reaction-oh-snap-im-done-10zi9Pi3iVtX7W)
+
+The other great benefit of JSON is the structure. It's just a
+dictionary. A set of key-value pairs. And accessing the values of a
+dictionary is *super easy* compared to transversing the nodes of a
+tag-based document. We do it all the time with Ruby hashes. Not only is
+it more efficient, but it's also a function that's native to the
+language, rather than needing to build or use a library for parsing XML.
```

Comment:
Fair enough. I think especially since we don't really show XML's metadata capabilities in the above example, keeping this simple makes good sense.